components/prism-ruby.js

```diff
@@ -75,6 +77,10 @@
 		'symbol': {
 			pattern: /(^|[^:]):[a-zA-Z_]\w*(?:[?!]|\b)/,
 			lookbehind: true
+		},
+		'function': {
+			pattern: /(def )[\w.]+/,
```

Comment:
I had a little fun with a Ruby REPL and found that this pattern will also happily match:

```ruby
my_function_def variable;
```

But at the same time, it doesn't match

```ruby
def   
extra_spaces_and_new_line(arg)
end
```

Changing the lookbehind group to `(\bdef\s+)` ought to solve these cases.

Reply:
Good catch. (Although I’ve never seen someone put a newline between `def` and the method name)

Reply:
Well, me either, but the compiler can handle it, so there is probably someone out there using it.